### PR TITLE
Select gpu

### PR DESCRIPTION
**Pull request by kushalkolar**
basically wrapper for https://github.com/pygfx/pygfx/pull/709 and https://github.com/pygfx/wgpu-py/pull/481

Adds a `fastplotlib.utils.gpu` module, and exposes `enumerate_adapters`, `select_adapter`, and `print_wgpu_report` at top level.

Example:
```python
import fastplotlib as fpl

fpl.select_adapter(fpl.enumerate_adapters()[2])
```

Also added corresponding info to docs: https://fastplotlib--491.org.readthedocs.build/en/491/user_guide/gpu.html#gpu-info

### CODE REVIEW

File: docs/source/user_guide/gpu.rst [hunk 1]
```diff
@@ -45,17 +50,255 @@ For example, on a Thinkpad AMD laptop with a dedicated nvidia GPU this returns::
     llvmpipe (LLVM 15.0.6, 256 bits) (CPU) on Vulkan
     AMD Radeon Graphics (rembrandt, LLVM 15.0.6, DRM 3.52, 6.4.0-0.deb12.2-amd64) (Unknown) on OpenGL
 
+In jupyter all the available adapters are also listed when ``fastplotlib`` is imported.
+
+You can get more detailed info on each adapter like this::
+
+    print([a.request_adapter_info() for a in fpl.enumerate_adapters()])
```
Comment (almarklein) [inline docs/source/user_guide/gpu.rst:57]: Similarly, the list of dicts being printed is not easy to read. Using json or pprint is much friendlier:
```suggestion
    import pprint

    for a in fpl.enumerate_adapters():
        pprint.pprint(a.request_adapter_info())
```

**Review by almarklein — COMMENTED** [review]
Nice! Just 2 small comments.

**Comment by kushalkolar** [comment]
Do you think it makes sense to expose this as a top level import? I don't think there's a use case for many more gpu related functions to create a "fpl.gpu" namespace.

File: docs/source/user_guide/gpu.rst [hunk 2]
```diff
@@ -19,24 +19,29 @@ This can happen under the following circumstances:
...
 
-You can view all GPUs that are available to ``WGPU`` like this::
+You can get a summary of all adapters that are available to ``WGPU`` like this::
 
-    import wgpu
+    import fastplotlib as fpl
 
-    for adapter in wgpu.gpu.enumerate_adapters():
-        print(adapter.summary)
+    print([a.summary for a in fpl.enumerate_adapters()])
```
[the remark this answers is not in the document]
Comment (kushalkolar) [reply docs/source/user_guide/gpu.rst:44]: right, I should stop writing docs after midnight :laughing:

**Comment by almarklein** [comment]
> Do you think it makes sense to expose this as a top level import? I don't think there's a use case for many more gpu related functions to create a "fpl.gpu" namespace.

In `pygfx` we deliberately do not make it part of the toplevel namespace, because its not something most users need. Maybe more important, it makes code less portable. But in pygfx we already had a namespace for advanced wgou-specific stuff (`gfx.renderers.wgpu`). Perhaps introducing a namespace just for this is a bit overkill ...

**Comment by kushalkolar** [comment]
Thanks! I like having your input since you're much much experienced in API dsdign 😄

**Pull request merged**